Src/Web/Web.Shared.Net/ApplicationInsightsHttpModule.cs

```diff
@@ -28,6 +28,13 @@ public ApplicationInsightsHttpModule()
             {
                 // The call initializes TelemetryConfiguration that will create and Intialize modules
                 TelemetryConfiguration configuration = TelemetryConfiguration.Active;
+
+                // Initialize publisher only in the first instance of the module
+                var result = Interlocked.Increment(ref mutex);
```

Comment:
ideally with this implementation you should be releasing instance you holding on `Dispose` and allow the next module to keep working.

I'm not sure how it will work when there are two modules and one of them for some reason will be "unloaded".